stumpy/cache.py

```diff
@@ -107,3 +107,24 @@ def _get_cache():
     site_pkg_dir = site.getsitepackages()[0]
     numba_cache_dir = site_pkg_dir + "/stumpy/__pycache__"
     return [f.name for f in pathlib.Path(numba_cache_dir).glob("*nb*") if f.is_file()]
+
+
+def _recompile():
+    """
+    Recompile all njit functions
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+    """
```

Comment:
We may add a note here that says `recompiling can overwrite the cache files`

Reply:
Can we verify that? Or what do we need to confirm this?

Reply:
@seanlaw 
Good question! I tried to check it in two ways. 

(1) I set the env variable `NUMBA_DEBUG_CACHE` to 1 to get more details about the operation regarding cache. I noticed that `<func>.recompile()` results in saving cache index/data files into the cache directory. 

(2) I tried to see if I can use those cache files to validate them. Didn't know how to do it! So, instead, I added `cache=True` to the njit decorator of `core._calculate_squared_distance`, and noticed that changing fastmath and recompiling function can result in a new output in the second run of script.

```
# temp.py

import warnings
import sys
import time

import numpy as np

from stumpy import core, cache

warnings.filterwarnings("ignore")

func = core._calculate_squared_distance

s = 3
QT = 1453258.774722079
μ_Q = -55.35781464461123
σ_Q = 650.912209452633
μ_T = -264.9540642227162
σ_T = 722.0717285148526
isconstant_Q = False
isconstant_T = False

# args = (s, QT, μ_Q, σ_Q, μ_T, σ_T, isconstant_Q, isconstant_T)
args = (s, QT, μ_T, σ_T, μ_Q, σ_Q, isconstant_T, isconstant_Q)


if sys.argv[1] == 'FirstRun':
    cache._enable()
print('output: ', func(*args))

if sys.argv[1] == 'FirstRun':
    new_fastmath_flags = func.targetoptions['fastmath'] - {'reassoc'}
    func.targetoptions['fastmath'] = new_fastmath_flags
    func.recompile()
    print('output: ', func(*args))

print('=' * 50)
```

And then one can run this script twice. `python  temp.py FirstRun` and then `python temp.py SecondRun`. The two outputs in the first run are different due the change in the precision. The second output of the first run matches with the output in the second run. This shows that the cache file contains the changes in the fastmath flags.

---

As a side, I was wondering how one can take advantage of the `cache._enable()`. It does enable caching and results in writing files into cache directory. But, how can they be used later if there is no `cache=True` in the njit signature?